src/signature.c

```diff
@@ -174,103 +174,68 @@ static void p11prov_sig_freectx(void *ctx)
 
 #define DER_SEQUENCE 0x30
 #define DER_OBJECT 0x06
-#define DER_NULL 0x05, 0x00
+#define DER_NULL 0x05
```

Comment:
Do not understand this change, the ASN.1 for DER Null is type 5 followed by a 0 byte.

In fact everywhere you used DER_NULL later on was followed by a ", 0" addition.

So why change this and manually add zeros in the definitions ?

Reply:
The following 0 byte is the Length field of the TLV. I changed this because:
 - `DER_SEQUENCE` and `DER_OBJECT` are Tag bytes only. So this makes `DER_NULL` consistent and be the Tag byte only.
 - It makes the Length fields explicit in the actual definition, and it's easier to validate the DER data on the byte array.
 - And it does not hide two bytes inside a macro. This enables validating the length calculations in the byte arrays.

For the length calculation reasons, I also changed the OID lengths to be the length of the named OID prefix variable. Instead of the expected sub entry.

Reply:
Ah, thinking of it this way it makes a lot of sense.
Thanks!